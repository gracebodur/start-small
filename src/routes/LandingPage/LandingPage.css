
.card {
    border-radius: .25rem;
    border-style: solid;
    margin: 60px auto;
    width:300px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    padding: 16px;
    text-align: center;
    border-width: 1px;
    border-color: transparent;
}

/* Float four columns side by side */
.column {
    float: left;
    width: 25%;
    padding: 0 10px;
  }

.row {margin: 0 -5px;}

/* Clear floats after the columns */
.row:after {
  content: "";
  display: table;
  clear: both;
}

/* Responsive columns */
@media screen and (max-width: 600px) {
  .column {
    width: 100%;
    display: block;
    margin-bottom: 20px;
  }
}

