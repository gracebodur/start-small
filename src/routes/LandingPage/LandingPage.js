import React, {Component} from 'react'
import SearchButton from '../../components/SearchButton/SearchButton'
import './LandingPage.css'

class LandingPage extends Component {
constructor(props) {
    super(props)
    this.state = {
      projectsData: [],
      results: [],
      isLoggedIn: false,
      isLoggedOut: false,
    }
  }

  componentDidMount() {
    fetch('./data.json')
      .then(res => res.json())
      .then(result => {
      const projects = result.map(item => {
        return item
      })
      this.setState({
        projectsData: projects
    })
  })
}
    render() {
        return(
            <> 
            <div >
                <header className="App-header">
                    <h1>Start small</h1>
                    <h3>Make a big difference!</h3>
                </header>
                <section>
                    <ul>
                        <li>Pick a classroom project for your friends to support.</li>
                        <li>Add a personal message.</li>
                        <li>See your impact!</li>
                    </ul>
                </section>
            </div>
                <div>
                    {this.state.projectsData.map(item => (
                        <div key={item.id} className='card'>
                        <section>
                            <h3>{item.schoolName}</h3>
                            <h3>{item.city}</h3>
                            <h3>{item.state}</h3>
                            <p >&#42;{item.fundingStatus}</p>
                            <p >{item.fulfillmentTrailer}</p>
                            <p>&#8212;{item.teacherName}</p>
                        <div>
                            <img 
                                src={item.imageURL}
                                alt='feature classroom'>
                            </img>
                        </div>
                        </section>
                       
                        <div>
                            <a 
                                href={item.fundURL}
                                target='_blank'
                                rel="noopener noreferrer">
                                Funding page link
                            </a>
                        </div>
                    </div>
                    ))}
                    <div>
                        <SearchButton />
                    </div>
                </div>
            </>
        )

    }
   
}

export default LandingPage