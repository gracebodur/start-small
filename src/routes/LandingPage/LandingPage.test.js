import React from "react";
import toJson from "enzyme-to-json";
import LandingPage from "./LandingPage";

describe(`LandingPage component`, () => {
  const props = {
    id: "a",
    name: "test-class-name"
  };

  it("renders a LandingPage by default", () => {
    const wrapper = <LandingPage />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the LandingPage given props", () => {
    const wrapper = <LandingPage {...props} />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
