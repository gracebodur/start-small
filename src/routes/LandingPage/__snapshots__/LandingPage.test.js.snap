// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component renders a LandingPage by default 1`] = `undefined`;

exports[`LandingPage component renders the LandingPage given props 1`] = `undefined`;
